src/collapse/vsprof.rs

```diff
@@ -0,0 +1,222 @@
+                        let number_of_calls = &self.stack[last].1;
+                        if prev_number_of_calls < *number_of_calls {
+                            self.stack[last].1 -= prev_number_of_calls;
+                        }
+                    }
+
+                    self.stack
+                        .push((function_name.to_string(), number_of_calls));
+                }
+            }
+        } else {
+            return invalid_data_error!("Unable to parse function name from line:\n{}", line);
+        }
+
+        Ok(())
+    }
+
+    // Store the current stack in `occurences`
+    fn write_stack(&self, occurrences: &mut Occurrences) {
+        if let Some(nsamples) = self.stack.last().map(|(_, n)| *n) {
+            if nsamples > 0 {
+                let functions: Vec<_> = self.stack.iter().cloned().map(|(f, _)| f).collect();
+                occurrences.insert(functions.join(";"), nsamples);
+            }
+        }
+    }
+}
+
+/// Gets the number from the start of the line. This can either be a number <1000, in which case the
+/// line doesn't contain double quotes, or the number can be >1000, in which case the line does
+/// contain double quotes. In both cases `line` may start with a leading comma, which will be
+/// ignored.
+fn get_next_number(line: &str) -> io::Result<(usize, &str)> {
+    // Trim the leading comma, if any
+    let line = line.strip_prefix(',').unwrap_or(line);
+
+    let mut remove_leading_comma = false;
+    let split = if let Some(line) = line.strip_prefix('"') {
+        remove_leading_comma = true;
+        line.split_once('"')
+    } else {
+        line.split_once(',')
+    };
+
+    if let Some((num, remainder)) = split {
+        // Remove any thousands separators which may be present
+        // TODO: If Visual Studio were to output numbers with a `.` as thousands separators, this
+        //       would fail
+        let num = num.replace(',', "");
+
+        if let Ok(num) = num.parse::<usize>() {
+            if remove_leading_comma {
+                // `remainder` still has a leading comma, because the number is >1000. We need to
+                // remove it so we are consistent regardless of whether the number was wrapped in
+                // double quotes or not.
+                if let Some(remainder) = remainder.strip_prefix(',') {
+                    return Ok((num, remainder));
+                }
+            }
+
+            return Ok((num, remainder));
+        }
+    }
+
+    invalid_data_error!("Invalid number in line:\n{}", line)
+}
+
+/// Some files may start with the <U+FEFF> character (zero width no-break space). This
+/// causes the call to `starts_with` to return false, which in this case isn't what we want.
+/// As this character has no influence on the rest of the file, we can safely ignore it.
+fn line_matches_start_line(line: String) -> bool {
+    line.trim().trim_start_matches('\u{feff}') == START_LINE
```

Comment:
```suggestion
    line.trim().trim_start_matches('\u{feff}').starts_with(START_LINE)
```
To handle the case where new fields are added in the future?